sncn_installer/install.sh

```diff
@@ -36,41 +38,47 @@ do_install () {
 
 do_setup_interfaces () {
   # Add udev rule
-  sudo rm -rf $UDEV_RULES_FILE
-  echo "KERNEL==\"EtherCAT[0-9]*\", MODE=\"0664\", GROUP=\"$ETHERCAT_USER_GROUP\"" | sudo tee --append $UDEV_RULES_FILE
+  sudo rm -rf ${UDEV_RULES_FILE}
+  echo "KERNEL==\"EtherCAT[0-9]*\", MODE=\"0664\", GROUP=\"${ETHERCAT_USER_GROUP}\"" | sudo tee --append ${UDEV_RULES_FILE}
 
   # Add detected interfaces and delete old ones
-  sudo sed -i '/MASTER.*DEVICE/d' /etc/sysconfig/ethercat
-  iface_index="0"
+  sudo sed -i '/MASTER._DEVICE/d' /etc/sysconfig/ethercat
   interfaces=$(ifconfig | grep -e "^e[tn][a-z0-9]*" -o)
 
   # Intel Up square 2
+  # 2nd ethernet is used for this board
   if [[ -n $(lscpu | sed '13q;d' | grep "Intel(R) Pentium(R) CPU N4200 @ 1.10GHz") ]];then
     MAC=$(cat /sys/class/net/enp3s0/address)
-    echo "MASTER${iface_index}_DEVICE=\"$MAC\"" | sudo tee --append $ETHERCAT_SYSCONFIG
+    echo "MASTER0_DEVICE=\"${MAC}\"" | sudo tee --append ${ETHERCAT_SYSCONFIG}
   else
   # Standard laptop
-    for iface in $interfaces ; do
-      MAC=$(cat /sys/class/net/$iface/address)
-      echo "MASTER${iface_index}_DEVICE=\"$MAC\"" | sudo tee --append $ETHERCAT_SYSCONFIG
-      iface_index=$((iface_index+1))
-    done
+    if [[ -n "${1}" ]];then
+      # Use interface passed as parameter if it is passed
+      iface="${1}"
+      MAC=$(cat /sys/class/net/${iface}/address)
+      echo "MASTER0_DEVICE=\"${MAC}\"" | sudo tee --append ${ETHERCAT_SYSCONFIG}
+    else
+      # Takes the first interface randomly if not
+      iface=$(${interfaces} | cut -d " " -f1)
+      MAC=$(cat /sys/class/net/${iface}/address)
+      echo "MASTER0_DEVICE=\"${MAC}\"" | sudo tee --append ${ETHERCAT_SYSCONFIG}
+    fi
   fi
-  sudo sed -i 's/DEVICE_MODULES=\"\"/DEVICE_MODULES=\"generic\"/g' $ETHERCAT_SYSCONFIG
+  sudo sed -i 's/DEVICE_MODULES=\"\"/DEVICE_MODULES=\"generic\"/g' ${ETHERCAT_SYSCONFIG}
 }
 
 do_start() {
-  if [[ -f "$ETHERCAT_INSTALL_PREFIX/etc/init.d/ethercat" ]]; then
-    sudo $ETHERCAT_INSTALL_PREFIX/etc/init.d/ethercat start
-  elif [[ -f "$ETHERCAT_START_PREFIX/etc/init.d/ethercat" ]]; then
-    sudo $ETHERCAT_START_PREFIX/etc/init.d/ethercat start
+  if [[ -f "${ETHERCAT_INSTALL_PREFIX}/etc/init.d/ethercat" ]]; then
+    sudo ${ETHERCAT_INSTALL_PREFIX}/etc/init.d/ethercat start
+  elif [[ -f "${ETHERCAT_INSTALL_PREFIX}/etc/init.d/ethercat" ]]; then
```

Comment:
What is the difference to the if branch?

Reply:
bad copy pasting ... Gonna correct it right now